### PR TITLE
Update update-a-style-block algorithm

### LINKED ISSUE #3179: document.implementation.createHTMLDocument breaks HTMLStyleElement.sheet
### Basic info:

- **Node.js version:** 14.6.0
- **jsdom version:** 16.5.3

Might be related to #1779

This bug does not occur when the document is the default document (`window.document`).

### Minimal reproduction case

```js
const { JSDOM } = require('jsdom');

const dom = new JSDOM();

const document = dom.window.document;
const styleDoc = document.implementation.createHTMLDocument('');
const styleElem = document.createElement('style');
styleElem.textContent = `.test { color: red; }`;
styleDoc.body.appendChild(styleElem);

console.log(styleElem.sheet); // null
```

### How does similar code behave in browsers?

`styleElem.sheet` is a CSSStyleSheet. See [example](https://jsbin.com/zajiwesaku/edit?js,console).

### PR DESCRIPTION
**Pull request by Qendolin**
The algorithm has changed, the style element is no longer required to have a browsing context. 

https://html.spec.whatwg.org/multipage/semantics.html#update-a-style-block

Fixes #3179

### CODE REVIEW

**Comment by Qendolin** [comment]
@TimothyGu can you review?

**Comment by Qendolin** [comment]
Sorry if I am being annoying but it has been over two weeks. 
@domenic

**Comment by domenic** [comment]
You are being annoying. We work on jsdom in our free time, which is rare, and I personally am experiencing some burnout on the project. This is not helpful.

**Comment by Qendolin** [comment]
Thank you for the response, if you don't have time then I don't mind waiting. I just wanted some confirmation that this PR is not being ignored or something.

**Review by domenic — APPROVED** [review]

**Review by domenic — CHANGES_REQUESTED** [review]
Can you test the contents of the sheet? Does it contain any rules?

**Review by domenic — APPROVED** [review]
LGTM as long as the test passes.

**Comment by domenic** [comment]
Hmm, yeah, this new test isn't passing. I'm glad I asked for it.

I'd suggest getting a dev environment set up that allows you to run the tests; then you should be able to figure out how to make them pass.

**Comment by Qendolin** [comment]
The code was already correct but not the test. I used the deprecated (but probably still used) `rules` property instead of `cssRules`. (I also noticed that JSDOM does not run any tests from `/css/cssom/` 🤔)

**Comment by domenic** [comment]
No, the test was correct. (It passed in Chrome and Firefox.) The code needs fixing.

**Comment by Qendolin** [comment]
The test cannot pass as long as [cssom's CSSStyleSheet](https://github.com/NV/CSSOM/blob/97bee0949153ecb3295eccf98da2be79b7269e2f/lib/CSSStyleSheet.js#L13-L16) does not support deprecated properties / methods. Or am I missing something?

**Comment by domenic** [comment]
It works elsewhere in jsdom...

**Comment by Qendolin** [comment]
That is not true, although `.rules` is used in many test, non of them are enabled.

This includes all tests at `test\web-platform-tests\tests\css\cssom\`, especially `test\web-platform-tests\tests\css\cssom\CSSStyleSheet.html`

How did I come to this conclusion? 
- I searched for `.rules` in all .html and .js test files.
- Looked at the folder, i.e.: `test\web-platform-tests\tests\css\cssom\`
- And ran `yarn test --fgrep css/cssom`
- None executed (only `css/cssom-view`)
- Add `DIR: css/cssom` in `to-run.yml`
- So many errors that I couldn't read them

You can also run this example:
```
const jsdom = require("jsdom");
const { JSDOM } = jsdom;

const dom = new JSDOM(`<!DOCTYPE html><style id="mystyle">p {color: red;}</style>`);

const sheet = dom.window.document.getElementById("mystyle").sheet
console.log("Rules: %o", sheet.rules);
console.log("Css Rules: %o", sheet.cssRules);
```

And you'll see that `sheet.rules` is undefined.

**Review by Klinter4802 — CHANGES_REQUESTED** [review]
#

**Comment by Qendolin** [comment]
If someone can provide me with an example of `.rules` working with JSDOM then I'ld be happy to fix my changes but as far as I can tell `.rules` never worked and all tests for it are disabled. 
If `.rules` is not supported in JSDOM and there are no other issues with my changes then please merge this PR. 
If you (you meaning contributes with write access) do not have time or do not intend to spend any more time on this PR then I'll close it.

**Pull request closed without merging**
